KnowYourWatts.ClientUI/ServerRequestHandler.cs

```diff
@@ -33,13 +37,91 @@ public class ServerRequestHandler(
             StandingCharge = standingCharge
         };
 
-        //initialReading += _randomisedValueProvider.GenerateRandomReading();
-
         await SendRequest(mpan, requestType, currentUsageRequest);
 
         return await HandleServerResponse(currentUsageRequest.CurrentReading);
     }
 
+    public async Task<string> GetPublicKey(string mpan)
+    {
+        // If mpan is null or whitespace, throw error as empty MPAN is invalid
+        if (string.IsNullOrWhiteSpace(mpan))
+        {
+            throw new ArgumentException("MPAN cannot be null or empty.");
```

Comment:
we shouldn't handle errors in this way, we should display an error and return an empty string